### DISCUSSION TITLE
Server Error Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

### DISCUSSION THREAD

**Discussion by Akhil7K** [prisma/prisma#22213] [Q&A] [answered]
### Question

I'm building a personal NextJS project and I'm using Prisma as an ORM and MySQL as my provider using MySQL workbench. Everything was well and fine in development stage of my project but after I deployed it to vercel I was facing errors as my database was inaccessible which I thought of as my db was hosted on my local machine, then i decided to use AWS RDS to host my db. I've created an instance in my local workbench which is connected to aws rds instance and I tried changing the database URL in my .env file with the updated URL and everything, but nothing seems to work for some reasons.  I'm unable to identify the issues and have been stuck at the same stage since then. Now, I'm facing another issue as when I'm trying to host my project on localhost, my Prisma is not initializing, I've tried doing Prisma generate and everything, and I cloned my repository from GitHub to revert all the changes but still nothing seems to fix the issue.

![Screenshot 2023-12-02 140000](https://github.com/prisma/prisma/assets/146330175/76958ba7-d3e2-4148-8b12-74f70d3999cf)
Error I'm getting:
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues

terminal error:
 

### Information about Prisma Schema, Client Queries and Environment (optional)

![Screenshot 2023-12-02 140110](https://github.com/prisma/prisma/assets/146330175/2ad9f36e-7f55-46d4-b9e0-ac6e757b089a)

![Screenshot 2023-12-02 140255](https://github.com/prisma/prisma/assets/146330175/f566e2ed-60cd-492d-b201-65b5111070e8)

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it
![Screenshot 2023-12-02 140430](https://github.com/prisma/prisma/assets/146330175/7c0ee5b6-2cfa-4331-8674-dad6dd9885e6)

**Comment by ludralph** [comment, marked as answer]
Hi @Akhil7K 👋 

Thank you for raising this question.

The error message you're seeing suggests that Prisma Client has not been generated or is not being recognized. This can happen if the `prisma generate` command is not run during the build process. To ensure that Prisma Client is generated during the build process, you can add the `prisma generate` command to the postinstall script in your package.json as shown in this [documentation](https://www.prisma.io/docs/guides/deployment/serverless/deploy-to-vercel#generate-prisma-during-build).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Akhil7K** [reply]
Hello @ludralph
I have already added `prisma generate` in my postinstall script, but for now I've been trying to run my project in my localhost, and these are the issues that arose. Although I've been able to resolve this issue, but a new one came up, and I've already raised another question about the same. Thank you for your time to review my issue.

**Reply by Pech66** [reply]
Hi @ludralph 👋

I have a problem with the Prisma client. I have already generated it, but it still shows the error. Even when I create a new project and do everything from scratch, it still shows the same error.

![image](https://github.com/user-attachments/assets/0f03414f-ac2c-4649-8204-a78681f7b737)

**Reply by luwukien** [reply]
> Hi @ludralph 👋
> 
> I have a problem with the Prisma client. I have already generated it, but it still shows the error. Even when I create a new project and do everything from scratch, it still shows the same error.
> 
> ![image](https://private-user-images.githubusercontent.com/176240068/457112085-0f03414f-ac2c-4649-8204-a78681f7b737.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA1Mjg1MTksIm5iZiI6MTc1MDUyODIxOSwicGF0aCI6Ii8xNzYyNDAwNjgvNDU3MTEyMDg1LTBmMDM0MTRmLWFjMmMtNDY0OS04MjA0LWE3ODY4MWY3YjczNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYyMVQxNzUwMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNjZiMDE5NDY2ODQ0M2Q4Y2NkYTkwMGI2ODgxZTM2NTFlYTNiMTg1MTBlMDgwNjdkYzRjY2UxMzI0YmI3YWFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.r-9Edkya-f7_5Q7YjF_iK3TR29n_sb3-N07emSOLnBM)

Hi, @Pech66

I guess that I same the issue with you. I fixed this issue by change output in schema.prisma into "../generated/prisma" and const { PrismaClient } = require("../generated/prisma")
. So you can try change location to solve this issue.
![image](https://github.com/user-attachments/assets/0e6afe3b-4be3-4aa4-ac64-dce50e6bdaac)

Reference: https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/generating-prisma-client#using-a-custom-output-path

**Comment by yashtech00** [comment]
"scripts": {  
  "dev": "next dev --port 3000 --turbopack",  
  "build": "npm run prisma:generate && next build",  
  "start": "next start",  
  "lint": "next lint --max-warnings 0",  
  "check-types": "tsc --noEmit",  
  "postinstall": "cd ../../packages/db && prisma generate",  
  "prisma:generate": "cd ../../packages/db && prisma generate" 
},bro i added but still i have same error did u find solution

**Comment by mochamadyudi** [comment]
I'm working with NestJS and encountered an issue where importing Prisma Client using:

```typescript
import { PrismaClient } from '@prisma/client';
```

did not work after running npx prisma generate.

Initially, after running ```npx prisma init```, the generator section looked like this:

```
generator client {
  provider = "prisma-client-js"
  output   = "../generated/prisma"
}

```

I then changed it to:

```typescript
generator client {
  provider        = "prisma-client-js"
  output          = "../node_modules/.prisma/client"
}
```

before this change, Prisma Client was only available via:
```typescript
import { PrismaClient } from 'generated/prisma';
```
But importing from ```@prisma/client``` still throws:
```typescript
Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
```

hope it helps you

**Reply by juniorknx** [reply]
-- npx prisma generate


prisma.service.ts

change for this:

import { Injectable, OnModuleInit } from '@nestjs/common';
import { PrismaClient } from '../generated/prisma/client';

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
  async onModuleInit() {
    await this.$connect();
  }
}

**Comment by marekdziem** [comment]
No matter what I try I still get @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.

WTF!?

**Reply by DaviLopesRodrigues** [reply]
1 - In your prisma.schema file, adjust the generator to: 
```
generator client {
  provider = "prisma-client-js"
  output = "../generated/prisma"
}.
```

2 - Run the `npx prisma generate` command

3 - In your prisma.service.ts file, adjust with the code snippet:
```
 import { OnModuleInit } from '@nestjs/common';
import { PrismaClient } from 'generated/prisma/client';

export class PrismaService extends PrismaClient implements OnModuleInit {
  async onModuleInit() {
    await this.$connect();
  }
}
```

**Reply by satyadalei** [reply]
Thank you.. !! DaviLopesRodrigues much appreciated 🙏.

**Comment by rhithesh** [comment]
import { PrismaClient } from "@/lib/generated/prisma"

this should work in newer next js versions hopefully this helps

**Comment by satyadalei** [comment]
The main issue is this line in your `schema.prisma`:

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "../generated/prisma"
}
```

By specifying a custom `output`, you’re telling Prisma to generate the client into `../generated/prisma`.
So when you run:

```bash
npx prisma generate
```

the Prisma Client will be created **inside that custom folder** instead of the default location.

---

### This is a problem, why?

* The `@prisma/client` npm package is specifically designed to hold the generated Prisma Client inside `node_modules/@prisma/client`.
* If you’re generating it into `../generated/prisma`, then **installing `@prisma/client` makes no sense**, because you won’t actually be using the one inside `node_modules`.
* This creates unnecessary confusion and inconsistency.

---

### Fix (Recommended)

Remove the custom `output` line completely:

```prisma
generator client {
  provider = "prisma-client-js"
}
```

This way, Prisma will correctly generate the client inside `node_modules/@prisma/client`.

👉 **Default & recommended → no custom output, use `@prisma/client`.**
Make sure to install it first:

```bash
npm install @prisma/client
```

Then run:

```bash
npx prisma generate
```

And finally, import Prisma Client the standard way:

```ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();
```

---

### Alternative (Custom Output)

If you **prefer to generate the client into a custom folder** (e.g., `../generated/prisma`), then:

* **Do not install** `@prisma/client`
* And do **not import** from `@prisma/client`

Instead, import directly from your custom path:

```ts
import { PrismaClient } from '../generated/prisma';

const prisma = new PrismaClient();
```

---

✅ In summary:

* **Default & recommended** → no custom output, use `@prisma/client`.
* **Custom setup** → generate into your own folder, but skip `@prisma/client` entirely.

---

**Reply by satyadalei** [reply]
@marekdziem this solution above might work..!

**Reply by dandilaminullah** [reply]
you are my savior homie
